pkg/compute/rust.go

```diff
@@ -148,6 +170,89 @@ func (r Rust) Verify(out io.Writer) error {
 
 	fmt.Fprintf(out, "Found Cargo.toml at %s\n", fpath)
 
+	// 5) Verify `fastly` crate version
+	//
+	// A valid and up-to-date version of the fastly crate is required.
+	fpath, err = filepath.Abs("Cargo.lock")
+	if err != nil {
+		return fmt.Errorf("error getting Cargo.lock path: %w", err)
+	}
+
+	if !common.FileExists(fpath) {
+		return fmt.Errorf("%s not found", fpath)
+	}
+
+	var lock CargoLock
+	if err := lock.Read("Cargo.lock"); err != nil {
+		return fmt.Errorf("error reading Cargo.lock file: %w", err)
+	}
+
+	// Fetch the latest crate version from cargo.io API.
+	l, err := GetLatestCrateVersion(r.client, "fastly")
+	if err != nil {
+		return err
+	}
+	latest, err := semver.Parse(l)
+	if err != nil {
+		return err
+	}
+
+	// Find the crate version declared in Cargo.lock lockfile.
+	var crate CargoPackage
+	for _, p := range lock.Package {
+		if p.Name == "fastly" {
+			crate = p
+			break
+		}
+	}
+
+	// If crate not found in lockfile, error with dual remediation steps.
+	if crate.Name == "" {
+		return errors.RemediationError{
+			Inner: fmt.Errorf("fastly crate not found"),
+			Remediation: fmt.Sprintf(
+				"To fix this error, edit %s with:\n\n\t %s\n\nAnd then run the following command:\n\n\t$ %s\n",
+				text.Bold("Cargo.toml"),
+				text.Bold(fmt.Sprintf(`fastly = "^%s"`, latest)),
+				text.Bold("cargo update -p fastly"),
+			),
+		}
+	}
+
+	// Parse lockfile version to semver.
+	version, err := semver.Parse(crate.Version)
+	if err != nil {
+		return fmt.Errorf("error parsing Cargo.lock file: %w", err)
+	}
+	// Parse the lowest minor semver for the latest release.
+	// I.e. v0.3.2 becomes v0.3.0.
+	latestMinor, err := semver.Parse(fmt.Sprintf("%d.%d.0", latest.Major, latest.Minor))
+	if err != nil {
+		return fmt.Errorf("error parsing Cargo.lock file: %w", err)
+	}
+
+	// If the lockfile version is within the minor range but lower than the
+	// latest, error with remediation to run `cargo update`.
+	if version.GTE(latestMinor) && version.LT(latest) {
+		return errors.RemediationError{
+			Inner:       fmt.Errorf("fastly crate not up-to-date"),
+			Remediation: fmt.Sprintf("To fix this error, run the following command:\n\n\t$ %s\n", text.Bold("cargo update -p fastly")),
+		}
+	}
+
+	// If version is on a lower minor or major, error with a dual remediation.
+	if version.LT(latest) {
+		return errors.RemediationError{
+			Inner: fmt.Errorf("fastly crate not up-to-date"),
+			Remediation: fmt.Sprintf(
+				"To fix this error, edit %s with:\n\n\t %s\n\nAnd then run the following command:\n\n\t$ %s\n",
+				text.Bold("Cargo.toml"),
+				text.Bold(fmt.Sprintf(`fastly = "^%s"`, latest)),
+				text.Bold("cargo update -p fastly"),
+			),
+		}
+	}
+
```

Comment:
Not a request for change, but I'm surprised this level of semantic version "parsing" is required by Verify...

Reply:
Haha me too 😆